checks/prgenv/cuda/cuda_aware_mpi.py

```diff
@@ -138,3 +139,129 @@ def set_num_tasks(self):
+        'MPICH_RDMA_ENABLED_CUDA': '1',
+        'MPICH_VERSION_DISPLAY': '1',
+        'LD_LIBRARY_PATH': '$CRAY_LD_LIBRARY_PATH:$LD_LIBRARY_PATH',
+    }
+    prebuild_cmds = [
+        'rm -fr MATLAB* series c++11_cuda',
+        'cd posts/cuda-aware-mpi-example/src'
+    ]
+    build_system = 'Make'
+    maintainers = ['@ekouts', '@jgphpc']
+    tags = {'production', 'scs', 'craype'}
+    cdt_info = parameter(find_cdts('daint:gpu', 'PrgEnv-gnu', 'cdt/'))
+    nvhpc_info = parameter(find_cdts('daint:gpu', 'PrgEnv-gnu',
+                                     'nvhpc-nompi/'))
+    gcc_info = parameter(find_cdts('daint:gpu', 'PrgEnv-gnu', 'gcc/'))
+    gpu_arch = variable(str, type(None))
+
+    @run_after('init')
+    def apply_module_info(self):
+        # bad_pe= ['cdt/21.09', 'cdt/20.08']
+        # making sure "gcc version" is compatible with "cuda version" in nvhpc,
+        # nvhpc/22.3 has cuda/11.6 which supports gcc<12:
+        nvhpc2gcc = {
+            '21.3': {'cuda': '11.2', 'gcc': '10'},
+            '21.5': {'cuda': '11.3', 'gcc': '10'},
+            '21.9': {'cuda': '11.4', 'gcc': '11'},
+            '22.2': {'cuda': '11.6', 'gcc': '11'},
+            '22.3': {'cuda': '11.6', 'gcc': '11'},
+            # TODO: newer nvhpc
+        }
+        gcc_major_version = self.gcc_info.split("/")[1].split(".")[0]
+        nvhpc_version = self.nvhpc_info.split("/")[1]
+        gcc_max_version = nvhpc2gcc[nvhpc_version]['gcc']
+        cuda_max_version = nvhpc2gcc[nvhpc_version]['cuda']
+        skip_msg = (
+            f'gcc/{gcc_major_version} != nvhpc/{nvhpc_version}:'
+            f'cuda/{cuda_max_version}:gcc/{gcc_max_version}'
+        )
+        self.skip_if(gcc_major_version != gcc_max_version, skip_msg)
+        self.modules = [self.cdt_info, self.nvhpc_info, self.gcc_info]
+
+    @run_before('compile')
+    def set_compilers(self):
+        gput = self.current_partition.select_devices('gpu')[0]
+        gcd_flgs = f'-arch={gput.arch}'
+        nvhpc_version = self.nvhpc_info.split("/")[1]
+        cuda_path = f'/opt/nvidia/hpc_sdk/Linux_x86_64/{nvhpc_version}/cuda'
+        link_flags = (
+            # add -lcuda to avoid segmentation fault
+            f'-Wl,-rpath={cuda_path}/lib64/ -lcuda '
+            # TODO: add -lmpi_gtl_cuda to avoid hangs (alps)
+            # '$PE_MPICH_GTL_DIR_nvidia80 $PE_MPICH_GTL_LIBS_nvidia80'
+        )
+        self.build_system.options = [
+            f'CUDA_INSTALL_PATH={cuda_path}',
+            'MPI_HOME=$CRAY_MPICH_PREFIX',
+            f'GENCODE_FLAGS="{gcd_flgs}"',
+            f'MPICC="{self.current_environ.cc}"',
+            f'MPILD="{self.current_environ.cxx} {link_flags}"',
+        ]
+
+    @run_before('compile')
+    def extract_versions(self):
+        cmd1 = "ldd ../bin/jacobi_cuda_aware_mpi | grep libcudart "
+        cmd2 = "awk '{print \"ls -l \",$3}'"
+        cmd3 = "sh"
+        cmd4 = "awk '{print $11}'"
+        self.rpt = os.path.join(self.stagedir, 'rpt')
```

Comment:
```suggestion
        rpt = os.path.join(self.stagedir, 'rpt')
```

Reply:
I need self.rpt in report_linked_versions below